File: source/extensions/TextHelper/TextHelper.java
package source.extensions.TextHelper;

import com.badlogic.gdx.graphics.g2d.BitmapFont;
import com.badlogic.gdx.graphics.g2d.GlyphLayout;
import com.badlogic.gdx.math.Vector2;

/**
 * @author Xico
 */

public class TextHelper {

    /**
     * Centers a piece of text in the screen and returns it's coordinates under a {@link Vector2}.
     *
     * @param font        {@link BitmapFont} used.
     * @param text        text to be centered.
     * @param worldWidth  width of the world (world coordinates).
     * @param worldHeight height of the world (world coorinates).
     * @return centered coordinates.
     */
    public static Vector2 centerOnScreen(BitmapFont font, String text, int worldWidth, int worldHeight) {
        final GlyphLayout layout = new GlyphLayout(font, text);
        final float x = (worldWidth - layout.width) / 2;
        final float y = (worldHeight - layout.height) / 2;
        return new Vector2(x, y);
    }

    /**
     * Centers a piece of text horizontally, given the world's width. The text's Y is defined by
     * the given parameter.
     *
     * @param font       {@link BitmapFont} used.
     * @param text       text to be centered.
     * @param y          text's target Y.
     * @param worldWidth width of the world (world coordinates).
     * @return centered coordinates.
     */
    public static Vector2 centerHorizontally(BitmapFont font, String text, float y, int worldWidth) {
        final GlyphLayout layout = new GlyphLayout(font, text);
        final float x = (worldWidth - layout.width) / 2;
        return new Vector2(x, y);
    }

    /**
     * Centers a piece of text inside an image.
     *
     * @param font      {@link BitmapFont} used.
     * @param text      text to be centered.
     * @param imgWidth  width of the image.
     * @param imgHeight height of the image.
     * @param imgPos    position of the image
     * @return centered coordinates.
     */
    public static Vector2 centerOnImage(BitmapFont font, String text, float imgWidth, float imgHeight, Vector2 imgPos) {
        final GlyphLayout layout = new GlyphLayout(font, text);
        final float x = imgPos.x + ((imgWidth - layout.width) / 2);
        final float y = imgPos.y + ((imgHeight / 2) + (layout.height / 2));
        return new Vector2(x, y);
    }
}

File: source/extensions/text/TextHelper.kt
package source.extensions.text

import com.badlogic.gdx.graphics.g2d.BitmapFont
import com.badlogic.gdx.graphics.g2d.GlyphLayout
import com.badlogic.gdx.math.Vector2

/**
 * Centers a piece of [text] on the screen, given the [worldWidth], [worldHeight] and [font] used.
 *
 * @param font        the [BitmapFont] used.
 * @param text        the [String] text to be centered.
 * @param worldWidth  width of the game world (world coordinates).
 * @param worldHeight height of the game world (world coordinates).
 * @return the correct, centered coordinates, in the form of a [Vector2].
 */
fun center(font: BitmapFont, text: String, worldWidth: Int, worldHeight: Int): Vector2 {
  val layout: GlyphLayout = GlyphLayout(font, text)
  val x: Float = (worldWidth - layout.width) / 2f
  val y: Float = (worldHeight + layout.height) / 2f
  return Vector2(x, y)
}

/**
 * Centers a piece of [text] horizontally on the screen, given the [worldWidth],
 * target [y] and [font] used.
 *
 * @param font        the [BitmapFont] used.
 * @param text        the [String] text to be centered.
 * @param worldWidth  width of the game world (world coordinates).
 * @param y           y value of the text's position.
 * @return the correct, centered coordinates, in the form of a [Vector2].
 */
fun centerX(font: BitmapFont, text: String, worldWidth: Int, y: Float): Vector2 {
  val layout: GlyphLayout = GlyphLayout(font, text)
  val x: Float = (worldWidth - layout.width) / 2f
  return Vector2(x, y)
}

/**
 * Centers a piece of [text] horizontally on the screen, given the [worldHeight],
 * target [x] and [font] used.
 *
 * @param font          the [BitmapFont] used.
 * @param text          the [String] text to be centered.
 * @param worldHeight   width of the game world (world coordinates).
 * @param x             x value of the text's position.
 * @return the correct, centered coordinates, in the form of a [Vector2].
 */
fun centerY(font: BitmapFont, text: String, worldHeight: Int, x: Float): Vector2 {
  val layout: GlyphLayout = GlyphLayout(font, text)
  val y: Float = (worldHeight + layout.height) / 2f
  return Vector2(x, y)
}

/**
 * Centers a piece of [text] inside an image of dimensions <width x height>, located at [position].
 *
 * @param font      the [BitmapFont] used.
 * @param text      the [String] text to be centered.
 * @param width     the width of the outer image.
 * @param height    the height of the outer image.
 * @param position  the position of the outer image, in the form of a [Vector2].
 * @return the correct, centered coordinates, in the form of a [Vector2].
 */
fun centerOnImage(font: BitmapFont, text: String, width: Float, height: Float, position: Vector2): Vector2 {
  val layout: GlyphLayout = GlyphLayout(font, text)
  val x: Float = position.x + ((width - layout.width) / 2f)
  val y: Float = position.y + ((height / 2f) + (layout.height / 2f))
  return Vector2(x, y)
}